lbrynet/core/Wallet.py

```diff
@@ -1371,9 +1393,26 @@ def _run_cmd_as_defer_to_thread(self, command_name, *args, **kwargs):
     def _update_balance(self):
         accounts = None
         exclude_claimtrietx = True
-        d = self._run_cmd_as_defer_succeed('getbalance', accounts, exclude_claimtrietx)
-        d.addCallback(
-            lambda result: Decimal(result['confirmed']) + Decimal(result.get('unconfirmed', 0.0)))
+
+        # Call doesn't always succeed as there is a race condition on startup
+        # Supply it with dummy data in the event of failure for nicer exception
+        # handling.
+        # See issue: https://github.com/lbryio/lbry/issues/626
+        try:
```

Comment:
This would read easier as an `inlineCallback`, and instead of catching this as an `AttributeError` you can use something like:
```
@defer.inlineCallbacks
def _update_balance(self):
    accounts = None
    exclude_claimtrietx = True
    balance = Decimal(0)
    if self.network and self.network.is_connected(): # See issue: https://github.com/lbryio/lbry/issues/626
        balance_info = yield self._run_cmd_as_defer_succeed('getbalance', accounts, exclude_claimtrietx)
        balance = Decimal(balance_info['confirmed']) + Decimal(balance_info.get('unconfirmed', 0.0))
    else:
        log.warning("lbryum is not connected")
    defer.returnValue(balance)

```

Reply:
I'll look into that, thanks